src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -10,11 +10,10 @@ import scala.math.Ordering.Implicits._
 import scala.reflect.ClassTag
 
 // interval inclusive of start, exclusive of end: [start, end)
-case class Interval[T](start: T, end: T)(implicit ev: Ordering[T]) extends Ordered[Interval[T]] {
-
+case class Interval[T](start: T, end: T)(implicit ev: Ordering[T]) extends Serializable {
   import ev._
 
-  require(start < end, s"invalid interval: $this: start is not before end")
+  require(start <= end, s"invalid interval: $this: start is not before end")
```

Comment:
This has tripped me up in the past, but I'm not sure it was necessarily wrong? Given your definition, `start == end` is a zero-length interval - has the definition changed, or is this now just allowed (and will return nothing)?

Reply:
Interval clearly expected to support zero-length intervals, see the definition of isEmpty:

 >   def isEmpty: Boolean = start == end

The definition of an interval [a, b) is all x such that a <= x and x < b.  Nothing in that definition precludes [7, 5), for example.  I intend to support such "improper" intervals in a later PR (I thought this one had already done enough).

I had to include the quality case here because the equality filter in Interval.gen causes the tests to fail in the case you try to construct examples of intervals over a type with only a single element (example: Interval[Interval[Boolean]], with the rule start < end, there is only one interval: [false, true)).

Reply:
Gotcha. From a user-perspective, it may be worth throwing a warning if you provide an empty interval (I know a lot of people do `vds.filter_intervals(Interval.parse('22:100-100'))` hoping to find their one variant - right now this fails explicitly, but soon it will fail silently).